client/dashboard/sites/settings-repositories/connect-repository-form.tsx

```diff
@@ -0,0 +1,437 @@
+		: [ { label: __( 'Select a branch' ), value: '' } ];
+
+	const { createDeployment } = useCreateCodeDeployment( site.ID, {
+		onSuccess: () => {
+			onConnected();
+		},
+		onError: () => {
+			setIsSubmitting( false );
+		},
+	} );
+
+	const handleAddGithubAccount = useCallback( () => {
+		const popup = window.open( INSTALL_APP_URL, '_blank', 'noopener' );
+
+		if ( ! popup ) {
+			return;
+		}
+
+		const handleFocus = () => {
+			refetchInstallations();
+			window.removeEventListener( 'focus', handleFocus );
+		};
+
+		window.addEventListener( 'focus', handleFocus );
+	}, [ refetchInstallations ] );
+
+	const handleSubmit = async () => {
+		if ( ! selectedRepository || ! selectedInstallation || ! branch || ! targetDir ) {
+			return;
+		}
+
+		setIsSubmitting( true );
+
+		try {
+			await createDeployment( {
+				externalRepositoryId: selectedRepository.id,
+				branchName: branch,
+				targetDir,
+				installationId: selectedInstallation.external_id,
+				isAutomated,
+				workflowPath,
+			} );
+		} catch ( error ) {
+			setIsSubmitting( false );
+		}
+	};
+
+	const installationOptions = useMemo( () => {
+		return [
+			{ label: __( 'Select an account' ), value: '' },
+			...installations.map( ( installation ) => ( {
+				label: installation.account_name,
+				value: installation.external_id.toString(),
+			} ) ),
+		];
+	}, [ installations ] );
+
+	const repositoryOptions = useMemo( () => {
+		return repositories.map( ( repo ) => ( {
+			label: `${ repo.owner }/${ repo.name }`,
+			value: repo.id.toString(),
+		} ) );
+	}, [ repositories ] );
+
+	const installationHelpText = useMemo( () => {
+		if ( isLoadingInstallations ) {
+			return __( 'Loading GitHub accounts…' );
+		}
+
+		if ( installationsError ) {
+			return __( 'We could not load your GitHub accounts. Try again after installing the app.' );
+		}
+
+		if ( installations.length === 0 ) {
+			return __( 'Add a GitHub account to select a repository.' );
+		}
+
+		return undefined;
+	}, [ installations, installationsError, isLoadingInstallations ] );
+
+	const repositoryHelpText = useMemo( () => {
+		if ( ! selectedInstallation ) {
+			return __( 'Select a GitHub account first.' );
+		}
+
+		if ( isLoadingRepositories ) {
+			return __( 'Loading repositories…' );
+		}
+
+		if ( repositories.length === 0 ) {
+			return __( 'No repositories available for this account.' );
+		}
+
+		return undefined;
+	}, [ isLoadingRepositories, repositories, selectedInstallation ] );
+
+	const isAdvancedValid = ! isAdvancedSelected || !! workflowPath;
+	const isFormValid = !! (
+		selectedRepository &&
+		selectedInstallation &&
+		branch &&
+		targetDir &&
+		isAdvancedValid
+	);
+
+	return (
+		<VStack spacing={ 6 }>
+			<VStack spacing={ 1 }>
+				<Text weight={ 600 }>{ __( 'Configure repository connection' ) }</Text>
+				<Text variant="muted">
+					{ __( 'Select a repository and choose where you’d like your files to deploy.' ) }
+				</Text>
+			</VStack>
+
+			<VStack spacing={ 2 }>
```

Comment:
We should use DataForm for all these so that it's DataForm that defines the stylings etc. 🙏